### ISSUE TITLE
package broken/ won't build

### ISSUE THREAD

**Issue by spl0i7** [elastic/go-sysinfo#48]
I use Go agent for elastic apm but and go-sysinfo is one of the dependency there but since this morning I am getting this error while trying to build my packages

```
github.com/elastic/go-sysinfo/providers/linux
src/github.com/elastic/go-sysinfo/providers/linux/host_linux.go:45:20: cannot convert filepath.Join(hostFS, procfs.DefaultMountPoint) (type string) to type procfs.FS
src/github.com/elastic/go-sysinfo/providers/linux/host_linux.go:64:42: h.procFS.Path undefined (type procfs.FS has no field or method Path)
src/github.com/elastic/go-sysinfo/providers/linux/process_linux.go:91:13: p.fs.Path undefined (type procfs.FS has no field or method Path)
```
help?

**Comment by BPMathews** [comment]
I am seeing the same issue. My builds are using Dep and it looks like the issue is a change made recently to https://github.com/prometheus/procfs.

**Comment by kientv** [comment]
Me too:
```# [project_name]/vendor/github.com/elastic/go-sysinfo/providers/linux
vendor/github.com/elastic/go-sysinfo/providers/linux/host_linux.go:45:20: cannot convert filepath.Join(hostFS, procfs.DefaultMountPoint) (type string) to type procfs.FS
vendor/github.com/elastic/go-sysinfo/providers/linux/host_linux.go:64:42: h.procFS.Path undefined (type procfs.FS has no field or method Path)
vendor/github.com/elastic/go-sysinfo/providers/linux/process_linux.go:91:13: p.fs.Path undefined (type procfs.FS has no field or method Path)```

**Comment by bouwerp** [comment]
I'm also having this issue. A workaround for applications utilising go modules could possibly be to just change to the last working version/commit. Any ideas for good old GOPATH applications?

We're planning on migrating to modules in the coming weeks, but unfortunately haven't started yet.

**Comment by andrewkroh** [comment]
In #49 this project was updated to be compatible with the new github.com/prometheus/procfs changes (while still working with the previous version).

**Comment by axw** [comment]
This has been fixed by https://github.com/elastic/go-sysinfo/pull/49. Your builds should proceed as normal now. I would recommend either using modules or vendoring to avoid future CI breakages.

**Comment by bouwerp** [comment]
Great - thanks @axw!

**Comment by kientv** [comment]
Do you need to increase release tag?

**Comment by axw** [comment]
@kientv do you have a need for that? For people using modules, the included go.mod hasn't been updated, so they'll be fine. For people using dep/glide/etc., I'm assuming prometheus/procfs is pinned as it does not have a stable API. Everyone else isn't using a package manager, so it doesn't matter :)

Due to the above, I think it's OK to defer creating a new release. Happy to hear arguments otherwise.

**Comment by EDDYCJY** [comment]
@axw We just happened to have this problem, we use dep to update apm which will cause prometheus/procfs to update...

**Comment by axw** [comment]
@EDDYCJY I see. Clearly I don't use dep ;)
I'll create a PR to bump go-sysinfo to 1.0.1.

**Comment by EDDYCJY** [comment]
Thanks, we are manually modifying one by one. Really a nightmare...

**Comment by kientv** [comment]
@axw  Thanks you.

**Comment by axw** [comment]
Just released [v1.0.1](https://github.com/elastic/go-sysinfo/releases/tag/v1.0.1).
